### PR TITLE
Sort pipeline

### PR DESCRIPTION
**Pull request by lizgzil**
Fixing both https://github.com/wellcometrust/nutrition-labels/issues/78 and https://github.com/wellcometrust/nutrition-labels/issues/86

Instead of a pipeline for each of the test, open and private runs, create one which takes a single config file as a variable.

- there are config files for test, open and private runs.
- the config files now contain everything for the whole pipeline - previously each step of the pipeline used it's own config.

Now they are in one config the variable names aren't the most ideal or descriptive and there are times where there are multiple variable names pointing to the same file , e.g.
```
grant_data_filedir = data/raw/wellcome-grants-awarded-2005-2019.csv
grants_text_data_file = data/raw/wellcome-grants-awarded-2005-2019.csv
```
but to change this would involve quite a bit overhaul of the various scripts - so I will leave it as is. I think there is definitely an improvement by using one config per pipeline run, so at least we have that!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -76,21 +76,33 @@ The code for this project is in the `nutrition_labels` and `notebooks` folder. M
 
 ### Pipeline overview <a name="pipelineoverview"></a>
 
+#### ResearchFish and EPMC evaluation data
+
+Create RF and EPMC evaluation data by running:
+
+`python nutrition_labels/create_training_data.py --config_path configs/training_data/2021.03.29.epmc.ini`
+and 
+`python nutrition_labels/create_training_data.py --config_path configs/training_data/2021.03.29.rf.ini`
```
Comment (aCampello) [inline README.md:85]: nice

File: configs/pipeline/2021.05.01.open.ini [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+grant_id_col = Internal ID
+# Variables used in evaluate.py
+epmc_file_dir = data/processed/training_data/210329epmc/training_data.csv
+rf_file_dir = data/processed/training_data/210329rf/training_data.csv
+eval_id_col = Internal ID
+eval_label_name = Relevance code
+
+[data_col_ranking]
+# Variables used in create_training_data.py
+epmc_col_ranking = ['Revised code']
```
Comment (aCampello) [inline configs/pipeline/2021.05.01.open.ini:36]: Maybe a comment to explain what are these? I couldn't figure out from the code.

**Review by aCampello — APPROVED** [review]
Good addition. Would be good to have some more comments, but no brainer if not!

File: configs/pipeline/2021.05.01.open.ini (hunk 2 above)
Comment (lizgzil) [reply configs/pipeline/2021.05.01.open.ini:36]: i've added something - it's a bit wordy but hopefully clearer!

File: configs/pipeline/2021.05.01.open.ini [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+rf_tags_filedir = data/raw/expanded_tags/research_fish_manual_edit.csv
+grant_tags_filedir = data/raw/expanded_tags/wellcome-grants-awarded-2005-2019_manual_edit_Lizadditions.csv
+prodigy_filedir = data/prodigy/merged_tech_grants/merged_tech_grants.jsonl
+grant_data_filedir = data/raw/wellcome-grants-awarded-2005-2019.csv
+sources_include = ['Prodigy grants', 'Grants']
+# Variables used in grant_tagger.py to train model
+training_data_file = data/processed/training_data/210501open/training_data.csv
+label_name = Relevance code
+training_data_file_id = Internal ID
+prediction_cols = ['Title', 'Description']
```
[the remark this answers is not in the document]
Comment (lizgzil) [reply configs/pipeline/2021.05.01.open.ini:19]: yeh not the most elegant! This is because these new config files are a combination of 4 different config files - so they were all designed with different scripts in mind and _then_ combined as one. There could certainly be more unity in these types of things if redesigned. In this case I wrote `create_training_data.py` in one way, and then `predict.py` and `evaluate.py` in another way.
